modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -128,5 +132,51 @@ object ConstructionService {
           }
         }
 
+    override def getPayloads(
+      operations: NonEmptyList[Operation],
+      maybeMetadata: Option[TransactionMetadata]
+    ): EitherT[F, ConstructionError, PayloadsResult] =
+      for {
+        _ <- EitherT.cond(operations.size == 2, (), IncorrectNumberOfOperations)
+
+        (positiveOperation, transactionAmount) <- EitherT.fromOption(
+          operations.collectFirst {
+            case op if op.amount.exists(!_.value.isNegative) =>
+              op.amount
+                .flatMap(_.value.toTransactionAmount.toOption)
+                .map((op, _))
+          }.flatten,
+          MissingPositiveAmountOperation
+        )
+
+        metadata <- EitherT.fromOption(maybeMetadata, MissingMetadata)
+
+        accountId <- EitherT.fromOption(positiveOperation.account, MissingAccountIdentifier)
+
+        dagTransaction =
+          DAGTransaction(
+            source = metadata.source,
+            destination = accountId.address,
+            amount = transactionAmount,
+            fee = metadata.fee,
+            parent = metadata.lastReference,
+            salt = metadata.salt.getOrElse(TransactionSalt(0L))
```

Comment:
if you want, it can be passed as a dependency into `ConstructionService`, but not necessarily as a `F[TransactionSalt]`. you can use the strategy pattern (or algebra/interpreter) to pass different implementation for different environments